src/Tests/dotnet-test.Tests/GivenDotnetTestBuildsAndRunsArtifactPostProcessing.cs

```diff
@@ -0,0 +1,249 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Xml;
+using System.Xml.Linq;
+using FluentAssertions;
+using Microsoft.DotNet.Tools.Test;
+using Microsoft.NET.TestFramework;
+using Microsoft.NET.TestFramework.Assertions;
+using Microsoft.NET.TestFramework.Commands;
+using Xunit;
+using Xunit.Abstractions;
+using CommandResult = Microsoft.DotNet.Cli.Utils.CommandResult;
+
+namespace Microsoft.DotNet.Cli.Test.Tests
+{
+    public class GivenDotnetTestBuildsAndRunsArtifactPostProcessing : SdkTest
+    {
+        private static object s_dataCollectorInitLock = new();
+        private static string s_dataCollectorDll;
+        private static string s_dataCollectorNoMergeDll;
+
+        public GivenDotnetTestBuildsAndRunsArtifactPostProcessing(ITestOutputHelper log) : base(log)
+        {
+            BuildDataCollector();
+            BuildDataCollectorNoMerge();
+        }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void ArtifactPostProcessing_CsProjs(bool merge)
```

Comment:
Could we try to have a bit more explanatory test names?

Reply:
The naming is the same of other tests...we defin csprojs when as argument for the `dotnet test` we have project files.

Reply:
Updated with `SolutionProjects`